windows/table.cpp

```diff
@@ -0,0 +1,275 @@
+#include "uipriv_windows.hpp"
+
+static void uiTableDestroy(uiControl *c);
+static void uiTableMinimumSize(uiWindowsControl *c, int *width, int *height);
+static BOOL onWM_NOTIFY(uiControl *c, HWND hwnd, NMHDR *nm, LRESULT *lResult);
+
+struct uiTable;
+
+struct uiTableModel {
+	uiTableModelHandler *mh;
+	std::vector<uiTable*> tables;
+};
+
+struct uiTableColumn {
+	uiTable *t;
+	WCHAR *name;
+	// don't really support parts (but this would part=>column mappings if we did)
+	int modelColumn;	// -1 = none
+};
+
+struct uiTable {
+	uiWindowsControl c;
+	uiTableModel *model;
+	HWND hwnd;
+	std::vector<uiTableColumn*> columns;
+};
+
+void *uiTableModelStrdup(const char *str)
+{
+	return strdup(str);
+}
+
+void *uiTableModelGiveColor(double r, double g, double b, double a)
+{
+	return 0;	// not implemented
+}
+
+uiTableModel *uiNewTableModel(uiTableModelHandler *mh)
+{
+	uiTableModel *m;
+
+	m = new uiTableModel();
+	m->mh = mh;
+	return m;
+}
+
+
+void uiFreeTableModel(uiTableModel *m)
+{
+	delete m;
+}
+
+void uiTableModelRowInserted(uiTableModel *m, int newIndex)
+{
+	LVITEM item;
+
+	item.mask = 0;
+	item.iItem = newIndex;
+	item.iSubItem = 0; //?
+	for (auto t : m->tables) {
+		ListView_InsertItem( t->hwnd, &item );
+	}
+}
+
+void uiTableModelRowChanged(uiTableModel *m, int index)
+{
+	for (auto t : m->tables) {
+		ListView_Update( t->hwnd, index );
+	}
+}
+
+void uiTableModelRowDeleted(uiTableModel *m, int oldIndex)
+{
+	for (auto t : m->tables) {
+		ListView_DeleteItem( t->hwnd, oldIndex );
+	}
+}
+
+void uiTableColumnAppendTextPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	uiTable *t = c->t;
+	int lvIndex = 0;
+
+	if (c->modelColumn >=0) {
+		return; // multiple parts not implemented
+	}
+	c->modelColumn = modelColumn;
+
+	// work out appropriate listview index for the column
+	for (auto candidate : t->columns) {
+		if (candidate == c) {
+			break;
+		}
+		if (candidate->modelColumn >= 0) {
+			++lvIndex;
+		}
+	}
+	
+	LV_COLUMN lvc;

```

Comment:
Consistency: declarations at the top

More important, though:

a) use `LVCOLUMNW` — `LV_COLUMN` is technically a deprecated name as of at least Vista if not earlier, and the `W` is consistency to explicitly use Unicode
b) you should call `ZeroMemory(&lvc, sizeof (LVCOLUMNW))` before setting any fields (partially consistency, partially correctness)